src/cls/hello/cls_hello.cc

```diff
@@ -258,6 +258,35 @@ static int bad_writer(cls_method_context_t hctx, bufferlist *in, bufferlist *out
 }
 
 
+class PGLSHelloFilter : public PGLSFilter {
+  string val;
+public:
+  PGLSHelloFilter(bufferlist::iterator& params) {
+    ::decode(xattr, params);
+    ::decode(val, params);
```

Comment:
Yes... since we're in the constructor here I think it might make sense to move the decode stuff into a separate ::load function or somesuch, so that people writing classes can clearly see how to raise an error.